c7n/resources/asg.py

```diff
@@ -1404,15 +1404,18 @@ def process_asg(self, asg):
              'Key': destination_tag,
              'Value': source['Value']}])
         if propagate:
```

Comment:
feels like this could be simplified into a one liner with `if propagate and asg['Instances']`

Reply:
That works. I had that first and tweaked it, should have stuck with the gut ;)